packages/client/hmi-client/src/utils/optimize.ts

```diff
@@ -138,63 +139,232 @@ export function createOptimizeChart(
 	};
 }
 
-export function createInterventionsChart(
-	data: { name: string; value: number; time: number; phase: string }[],
-	endTime: number
-): any {
-	const interventionName = data[0].name;
-	return {
-		$schema: 'https://vega.github.io/schema/vega-lite/v5.json',
-		description: 'A line chart with annotations.',
-		data: {
-			values: data
-		},
-		config: {
-			legend: { title: null, orient: 'top', direction: 'horizontal' }
-		},
-		layer: [
-			{
-				mark: { type: 'line', interpolate: 'step-after' },
-				width: 400,
-				height: 400,
-				encoding: {
-					x: {
-						field: 'time',
-						type: 'quantitative',
-						axis: { title: 'Time (days)', grid: false },
-						scale: { domain: [0, endTime] }
-					},
-					y: {
-						field: 'value',
-						type: 'quantitative',
-						axis: {
-							title: interventionName,
-							grid: true,
-							gridColor: { condition: { test: 'datum.value === 0', value: 'black' }, value: null }
-						},
-						scale: { padding: 10 }
-					},
-					color: {
-						field: 'phase',
-						type: 'nominal',
-						scale: { range: ['#a3a3a3', '#1B8073'] },
-						legend: { title: null },
-						sort: ['Before optimization', 'After optimization']
-					}
-				}
-			},
-			...createInterventionChartMarkers(data)
-		]
-	};
-}
-
-function createInterventionChartMarkers(
-	data: { name: string; value: number; time: number; phase: string }[]
-) {
+function createInterventionChartMarkers(data: { name: string; value: number; time: number }[]) {
 	return data.map((ele) => ({
+		data: [{}], // Dummy data to ensure the layer is rendered
 		mark: { type: 'rule', strokeDash: [4, 4], color: 'black' },
 		encoding: {
 			x: { datum: ele.time }
 		}
 	}));
 }
+
+export const createOptimizeForecastChart = (
+	preSampleRunData: Record<string, any>[],
+	preStatisticData: Record<string, any>[],
+	postSampleRunData: Record<string, any>[],
+	postStatisticData: Record<string, any>[],
+	interventionsData: { name: string; value: number; time: number }[],
+	options: any
```

Comment:
Would be good have typings established since it sort of drives the logic here.

Also a small note you accommodate an options.title, but you went with a custom "<h6>" title.

Reply:
Ill add the typings.  And thanks for the note about the title, will add